lib/archethic_web/live/transaction_details_live.ex

```diff
@@ -99,28 +105,69 @@ defmodule ArchethicWeb.TransactionDetailsLive do
        ) do
     previous_address = Transaction.previous_address(tx)
 
-    inputs = Archethic.get_transaction_inputs(address)
-    ledger_inputs = Enum.reject(inputs, &(&1.type == :call))
-    contract_inputs = Enum.filter(inputs, &(&1.type == :call))
-    uco_price_at_time = tx.validation_stamp.timestamp |> OracleChain.get_uco_price()
-    uco_price_now = DateTime.utc_now() |> OracleChain.get_uco_price()
+    timestamp = tx.validation_stamp.timestamp
 
-    token_properties =
-      get_token_addresses([], ledger_inputs)
-      |> get_token_addresses(transaction_movements)
-      |> get_token_addresses(token_transfers)
-      |> Enum.uniq()
-      |> get_token_properties()
+    async_assign_uco_price(timestamp)
+    async_assign_inputs_and_token_properties(address, transaction_movements, token_transfers)
 
     socket
     |> assign(:transaction, tx)
     |> assign(:previous_address, previous_address)
-    |> assign(:inputs, ledger_inputs)
-    |> assign(:calls, contract_inputs)
     |> assign(:address, address)
-    |> assign(:uco_price_at_time, uco_price_at_time)
-    |> assign(:uco_price_now, uco_price_now)
-    |> assign(:token_properties, token_properties)
+    |> assign(:uco_price_at_time, 0.0)
+    |> assign(:uco_price_now, 0.0)
+    |> assign(:inputs, [])
+    |> assign(:calls, [])
+    |> assign(:token_properties, %{})
+  end
+
+  defp async_assign_uco_price(timestamp) do
```

Comment:
I think this one is not necessary to use async here as it's a direct read of the ETS table.

Reply:
ok thanks, so I will keep everything inside handle_transaction function, except the 3 fields related to tokens and inputs.
do you think it's useful to first process inputs send a msg and assign and then spawn another task for tokens or it's not worth it ?

Reply:
I think it might, because the token information is a "good to have" but not a "must to have", whereas the list of inputs is a hard requirement.

Reply:
done